<div class="col-xs-12 col-sm-9 col-md-8 main-content">
 <section class="multi-line-signature">
  <span class="returntype">
   <a class="" href="dart-async/AsyncError-class.html">
    AsyncError
   </a>
  </span>
  <span class="name ">
   errorCallback
  </span>
  (
  <wbr>
  <span class="parameter" id="errorCallback-param-error">
   <span class="type-annotation">
    <a class="" href="dart-core/Object-class.html">
     Object
    </a>
   </span>
   <span class="parameter-name">
    error
   </span>
  </span>
  ,
  <span class="parameter" id="errorCallback-param-stackTrace">
   <span class="type-annotation">
    <a class="" href="dart-core/StackTrace-class.html">
     StackTrace
    </a>
   </span>
   <span class="parameter-name">
    stackTrace
   </span>
  </span>
  )
 </section>
 <section class="desc markdown">
  <p>
   Intercepts errors when added programmtically to a
   <code>
    Future
   </code>
   or
   <code>
    Stream
   </code>
   .
  </p>
  <p>
   When caling
   <a class="" href="dart-async/Completer/completeError.html">
    Completer.completeError
   </a>
   ,
   <a class="" href="dart-async/Stream-class.html">
    Stream.addError
   </a>
   ,
or
   <a class="" href="dart-async/Future-class.html">
    Future
   </a>
   constructors that take an error or a callback that may throw,
the current zone is allowed to intercept and replace the error.
  </p>
  <p>
   When other libraries use intermediate controllers or completers, such
calls may contain errors that have already been processed.
  </p>
  <p>
   Return
   <code>
    null
   </code>
   if no replacement is desired.
The original error is used unchanged in that case.
Otherwise return an instance of
   <a class="" href="dart-async/AsyncError-class.html">
    AsyncError
   </a>
   holding
the new pair of error and stack trace.
If the
   <a class="" href="dart-async/AsyncError/error.html">
    AsyncError.error
   </a>
   is
   <code>
    null
   </code>
   , it is replaced by a
   <a class="" href="dart-core/NullThrownError-class.html">
    NullThrownError
   </a>
   .
  </p>
 </section>
 <section class="summary source-code" id="source">
  <h2>
   <span>
    Source
   </span>
  </h2>
  <pre>
   <code class="prettyprint lang-dart">
    AsyncError errorCallback(Object error, StackTrace stackTrace);
   </code>
  </pre>
 </section>
</div>
